cove/lib/common.py

```diff
@@ -342,3 +371,110 @@ def get_json_data_deprecated_fields(schema_name, schema_url, json_data):
         deprecated_fields_output[field] = {"paths": slashed_paths, "explanation": paths[1]}
 
     return deprecated_fields_output
+
+
+def _get_schema_codelist_paths(schema_name, schema_url, obj=None, current_path=(), codelist_paths=None):
+    '''Get a dict of codelist paths including the filename and if they are open.
+
+    codelist paths are given as tuples of tuples:
+        {("path", "to", "codelist"): (filename, open?), ..}
+    '''
+    if codelist_paths is None:
+        codelist_paths = {}
+
+    if schema_url:
+        obj = get_schema_data(schema_url, schema_name)
+
+    for prop, value in obj['properties'].items():
+        if current_path:
+            path = current_path + (prop,)
+        else:
+            path = (prop,)
+
+        if "codelist" in value and path not in codelist_paths:
+            codelist_paths[path] = (value['codelist'], value.get('openCodelist', False))
+
+        if value.get('type') == 'object':
+            _get_schema_codelist_paths(None, None, value, path, codelist_paths)
+        elif value.get('type') == 'array' and value['items'].get('properties'):
+            _get_schema_codelist_paths(None, None, value['items'], path, codelist_paths)
+
+    return codelist_paths
+
+
+@functools.lru_cache()
+def _load_codelists(codelist_url, unique_files):
+    codelists = {}
+    for codelist_file in unique_files:
+        try:
+            response = requests.get(codelist_url + codelist_file)
+            reader = csv.DictReader(line.decode("utf8") for line in response.iter_lines())
+            codelists[codelist_file] = {}
+            for record in reader:
+                code = record.get('Code') or record.get('code')
+                title = record.get('Title') or record.get('Title_en')
+                codelists[codelist_file][code] = title
+
+        except requests.exceptions.RequestException:
+            raise
+
+    return codelists
+
+
+def _generate_data_path(json_data, path=()):
```

Comment:
I think we should use the same approach for deprecated paths in the data so as to avoid creating a fresh copy of the data in memory (thus addressing  issue #579)